File: CS/PersistentClasses.cs
using System;
using DevExpress.Xpo;

namespace DXSample {
    public class Order :XPObject {
        public Order(Session session) : base(session) { }

        private string fOrderName;
        public string OrderName {
            get { return fOrderName; }
            set { SetPropertyValue<string>("OrderName", ref fOrderName, value); }
        }

        private DateTime fOrderDate;
        public DateTime OrderDate {
            get { return fOrderDate; }
            set { SetPropertyValue<DateTime>("OrderDate", ref fOrderDate, value); }
        }
    }
}

File: VB/PersistentClasses.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace DXSample
	Public Class Order
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private fOrderName As String
		Public Property OrderName() As String
			Get
				Return fOrderName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("OrderName", fOrderName, value)
			End Set
		End Property

		Private fOrderDate As DateTime
		Public Property OrderDate() As DateTime
			Get
				Return fOrderDate
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue(Of DateTime)("OrderDate", fOrderDate, value)
			End Set
		End Property
	End Class
End Namespace